homeassistant/components/invoxia/const.py

```diff
@@ -0,0 +1,57 @@
+"""Constants for the Invoxia (unofficial) integration."""
+from __future__ import annotations
+
+from collections.abc import Mapping
+from datetime import timedelta
+import logging
+
+from gps_tracker.client.datatypes import TrackerIcon
+
+ATTRIBUTION = "Data provided by an unofficial client for Invoxia API."
+
+CLIENT = "client"
+
+DATA_UPDATE_INTERVAL = timedelta(seconds=240)
+
+DOMAIN = "invoxia"
+
+LOGGER = logging.getLogger(__package__)
+
+MDI_ICONS: Mapping[TrackerIcon, str] = {
```

Comment:
I've tried to use them but for now I do not get the behaviour I am expecting.

Invoxia app allows users to define a specific icon for each device. I would like to use the corresponding icon for the entity in home-assistant to offer a coherent experience between Invoxia app and home-assistant.

However, by default, home assistant will show an icon which indicates the `STATE_HOME` or `STATE_NOT_HOME` state.

I tried adding an extra state attribute named `icon_name` which indicates the icon returned by Invoxia API (e.g. `moto`, `car`, `dog`, ...) and I defined a icon translation file to map those states to their corresponding MDI icons, but this does not change the entity icon.

Is there a way to indicate that icons should be deduced from the "translated" attribute `icon_name` instead of using the default device tracker one?
Otherwise I can either keep using the `icon` attribute with my `MDI_ICONS` mapping, or I can use the default home-assistant behaviour and users will have to manually set the icon for their device on both invoxia and home-assistant which is only a slight inconvenience.
Either of these options is fine for me.

Reply:
The icon is not related to the state.
It is set (usually once) by the user on Invoxia Smartphone app to easily identify the device purpose (whether the tracker is in a car/on a bike/on the collar of a dog/...). Therefore we can consider the icon as "fixed" (though the user can change from time to time if they want to, but it's still a manual action, its not related to the entity state)

In its current implementation, this integration simply uses that per-device user-defined icon and set it on the entity but I could also set no icon at all and let the user manually define an icon in home-assistant if they do not want the default present/absent icon.